sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ExpressionTypeCheckingSuite.scala

```diff
@@ -745,4 +745,52 @@ class ExpressionTypeCheckingSuite extends SparkFunSuite with SQLHelper with Quer
       )
     )
   }
+
+  test("check types for Lag") {
+    val lag = Lag(Literal(1), NonFoldableLiteral(10), Literal(null), true)
+    assert(lag.checkInputDataTypes() ==
```

Comment:
Could you trigger the error from user space, please? This check is pretty trivial.

Reply:
Actually, I have added another UT from the perspective of user-facing in the '[DataFrameWindowFunctionsSuite.scala](https://github.com/apache/spark/pull/38520/files#diff-a7bdc96518ada8a7be55fe02d5fcc0080a4fff8e09819f87188d02edba9f4dc2)' for 'FRAME_LESS_OFFSET_WITHOUT_FOLDABLE' error class.
<img width="532" alt="image" src="https://user-images.githubusercontent.com/15246973/200162463-22e2c622-170e-4e41-8740-fb828e628411.png">